openscenario/openscenario_interpreter/include/openscenario_interpreter/simulator_core.hpp

```diff
@@ -66,10 +68,39 @@ class SimulatorCore
   class CoordinateSystemConversion
   {
   protected:
+    static auto checkRoutingAlgorithm(const RoutingAlgorithm::value_type & routing_algorithm)
+    {
+      std::set<RoutingAlgorithm::value_type> supported = {
+        RoutingAlgorithm::value_type::shortest, RoutingAlgorithm::value_type::undefined};
+
+      if (supported.find(routing_algorithm) == supported.end()) {
+        std::unordered_map<RoutingAlgorithm::value_type, std::string> name_map = {
+          {RoutingAlgorithm::value_type::assigned_route, "assignedRoute"},
+          {RoutingAlgorithm::value_type::fastest, "fastest"},
+          {RoutingAlgorithm::value_type::least_intersections, "leastIntersections"},
+          {RoutingAlgorithm::value_type::shortest, "shortest"},
+          {RoutingAlgorithm::value_type::undefined, "undefined"}};
+        std::stringstream what;
+        what << "There was an operation to calculate relative lane position with RoutingAlgorithm "
+                "set to "
+             << name_map[routing_algorithm] << ", but this is currently not supported. ";
+        what << "Please set RoutingAlgorithm to either 'shortest' or 'undefined'.";
+        throw common::Error(what.str());
+      }
```

Comment:
Checking to see if the RoutingAlgorithm is a supported one is not a task that should be done here.
It is the constructor of the RoutingAlgorithm class that does that. The RoutingAlgorithm::value_type stringification should also be done by the RoutingAlgorithm class, which should implement the stream output operator. (See, for example, the DynamicsShape class.)

Reply:
@yamacir-kit 
I completely agree with this. 
Such an implementation already exists and there are more types supported (see [routing_algorithm.cpp](https://github.com/tier4/scenario_simulator_v2/blob/560cb0538f139c38da9f59df2dcd896ae14934d7/openscenario/openscenario_interpreter/src/syntax/routing_algorithm.cpp#L55-L70)). 

This check appears in `SimulatorCore` only in `makeNativeRelativeLanePosition()` and `makeNativeBoundingBoxRelativeLanePosition()` - which are called in `DistanceCondition` and `RelativeDistanceCondition`. 
Therefore, I assume that in general `RoutingAlghoritm` is supported in a larger scope (5 types, e.g. `"fastest"`). However, for `DistanceCondition` and `RelativeDistanceCondition` only the `"shortest"` and `"undefined"` types are supported. 

To put it another way, this check is added only for `makeNativeRelativeLanePosition()` and `makeNativeBoundingBoxRelativeLanePosition()` calls, since just for them the rest of `RoutingAlghoritm` is not supported (e.g. `"fastest"`) - which does not mean that other scenario elements do not support it.

I would also add that this check was introduced not long ago as a result of PR: ["Implement DistanceCondition / RelativeDistanceCondition for shortest of RoutingAlgorithm"](https://github.com/tier4/scenario_simulator_v2/pull/1216) (previously only “undefined” was supported). 

In this PR, by separating this method I was only trying to “avoid repetitions”. Now I also see that making a temporary map is unnecessary - thanks so much for pointing out the ` operator<< ` overload.
I believe that this check can remain in `SimulatorCore`, or I can propose to move it to DistanceCondition and `RelativeDistanceCondition` - since for these elements only 2 types of `RoutingAlghoritm` are supported.

Please let me know what you think.

Reply:
> I believe that this check can remain in SimulatorCore, or I can propose to move it to DistanceCondition and RelativeDistanceCondition - since for these elements only 2 types of RoutingAlghoritm are supported.

Move it to the constructors of DistanceCondition and RelativeDistanceCondition. This is not the purpose of this task, but it is a good opportunity to address it now.

Note: use boost::lexical_cast to stringify objects via `operator <<`.